app/javascript/src/Common/PlanSelector.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { render } from 'react-dom'
+import { TextInput } from '@patternfly/react-core'
+
+import './PlanSelector.scss'
+
+class PlanSelector extends React.Component {
+
+  state = {
+    searchTerm: '',
+    showPlans: false,
+    plans: this.props.plans,
+    selectedPlan: { }
+  }
+
+  onTextChange = (searchTerm) => {
+    const filteredPlans = this.props.plans.filter(plan => plan.name.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1)
+    this.setState({ searchTerm, plans: filteredPlans })
+  }
+
+  onFocus = () => this.setState({ showPlans: true })
+
+  onBlur = () => this.setState({
+    searchTerm: this.state.selectedPlan.name || '',
+    showPlans: false,
+    plans: this.props.plans
+  })
+
+  onPlanSelected = (selectedPlan) => this.setState({
+    showPlans: false,
+    selectedPlan
+  })
+
+  renderPlans () {
+    const { plans } = this.state
+
+    const options = plans.length
+      ? plans.map(plan => <li key={plan.id} onMouseDown={() => this.onPlanSelected(plan)}>{plan.name}</li>)
```

Comment:
How this handles hundreds of plans that some customers can have?

Reply:
yes good catch, it's still in the making.